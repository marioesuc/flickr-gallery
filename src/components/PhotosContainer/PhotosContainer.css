.PhotosContainer-container {
  margin: 20px;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}
