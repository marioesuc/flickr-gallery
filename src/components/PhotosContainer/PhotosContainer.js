import React from 'react';
import './PhotosContainer.css';
import Photo from '../Photo/Photo';

// Custom PhotosContainer component
class PhotosContainer extends React.Component {
  handleOnPhotoClick = this.props.onPhotoClick;

  render() {
    const { data } = this.props;

    return (
      <div className='PhotosContainer-container'>
        {data.map(photo => {
          return (
            <Photo
              onClick={this.handleOnPhotoClick}
              key={photo.id}
              photoId={photo.id}
              url={`https://farm${photo.farm}.staticflickr.com/${
                photo.server
              }/${photo.id}_${photo.secret}_m.jpg`}
              title={photo.title}
            />
          );
        })}
      </div>
    );
  }
}

export default PhotosContainer;
