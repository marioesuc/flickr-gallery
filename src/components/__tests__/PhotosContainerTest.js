import React from 'react';
import { shallow, configure } from 'enzyme';
import PhotosContainer from '../PhotosContainer/PhotosContainer';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('PhotosContainer', () => {
  const props = {
    data: [
      {
        id: 'id',
        farm: 'farm',
        server: 'server',
        secret: 'secret',
        title: 'title'
      }
    ],
    onPhotoClick: jest.fn()
  };

  it('renders correctly', () => {
    const wrapper = shallow(<PhotosContainer {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
