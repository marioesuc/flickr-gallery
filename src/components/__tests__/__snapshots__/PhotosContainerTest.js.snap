// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotosContainer renders correctly 1`] = `ShallowWrapper {}`;
